scripts/ligandtocore/QD.py

```diff
@@ -0,0 +1,205 @@
+
+    # Imports the cores and ligands
+    core_list = qd_scripts.read_mol(core_folder, arg['input_cores'], arg['column'], arg['row'],
+                                    arg['smiles_extension'])
+    ligand_list = qd_scripts.read_mol(ligand_folder, arg['input_ligands'], arg['column'],
+                                      arg['row'], arg['smiles_extension'])
+
+    # Return the indices of the core dummy atoms
+    core_indices = [prep_core(core, core_folder, arg['dummy'], arg['core_opt']) for
+                    core in core_list]
+
+    # Open the ligand database and check if the specified ligand(s) is already present
+    if arg['use_database']:
+        database = qd_scripts.read_database(ligand_folder, arg['database_name'])
+    else:
+        database = [[], [], [], [], []]
+
+    # Rotate all the ligands and format the resulting list
+    ligand_list = [prep_ligand(ligand, ligand_folder, database, arg['ligand_opt'], arg['split'])
+                   for ligand in ligand_list]
+    ligand_list, ligand_indices, database_entries = zip(*ligand_list)
+    ligand_indices = list(itertools.chain(*ligand_indices))
+    ligand_list = itertools.chain(*ligand_list)
+
+    # Write new entries to the ligand database
+    if arg['use_database']:
+        qd_scripts.write_database(database_entries, ligand_folder, database)
+
+    # Combine the core with the ligands, yielding qd, and format the resulting list
+    qd_list = [prep_qd(core, ligand, core_indices[i], ligand_indices[j], qd_folder) for i, core in
+               enumerate(core_list) for j, ligand in enumerate(ligand_list)]
+    qd_list, pdb_name_list, qd_indices = zip(*qd_list)
+
+    # Check if the ADF environment variables are set and optimize the qd with the core frozen
+    if arg['qd_opt']:
+        sys_var = qd_scripts.check_sys_var()
+        if sys_var:
+            for i, qd in enumerate(qd_list):
+                qd_scripts.prep_ams_job(qd, pdb_name_list[i], qd_folder,
+                                        qd_indices[i], arg['maxiter'])
+
+    # The End
+    time_end = time.time()
+    print('\nTotal elapsed time:\t\t' + '%.4f' % (time_end - time_start) + ' sec')
+
+
+
+
+arguments = {
+    'input_cores': 'Cd68Se55.xyz',
+    'input_ligands': ['OCCCCCCCCC'],
+    'path': r'/Users/basvanbeek/Documents/CdSe/Week_5',
+    'dir_name_list': ['core', 'ligand', 'QD'],
+    'smiles_extension': '.txt',
+    'column': 0,
+    'row': 0,
+    'dummy': 'Cl',
+    'database_name': 'ligand_database.txt',
+    'use_database': True,
+    'core_opt': False,
+    'ligand_opt': True,
+    'qd_opt': False,
+    'maxiter': 10000,
+    'split': True
+}
+
+argv = [item for item in sys.argv[1:]]
+if argv:
+    argv = [item.split('=') for item in argv]
+    for item in argv:
+        arguments[item[0]] = item[1]
+
+
+# Runs the script: add ligand to core and optimize (UFF) the resulting qd with the core frozen
+prep_prep(**arguments)
```

Comment:
@BvB93 you can call the arguments as a simple dict like:
```
prep_prep(arguments)
```

and change the function definition like:
```
def prep_prep(args)
```
There is not need for the key-worded (**) notation